From: Thomas Renninger <trenn@suse.de>
Subject: Fix reboot issue with machines that do not have an i8042
References: bnc#703937
Patch-Mainline: yes and no (see below)

Mainline switched from BOOT_KBD to BOOT_ACPI unconditionally.
It's expected that this will cause regressions, therefore
BOOT_ACPI is only chosen by default if the ACPI FADT table
tells us that the system does not have an i8042 controller.

---
 arch/x86/kernel/reboot.c |    6 ++++++
 1 file changed, 6 insertions(+)

Index: linux-2.6.32-SLE11-SP2/arch/x86/kernel/reboot.c
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/arch/x86/kernel/reboot.c
+++ linux-2.6.32-SLE11-SP2/arch/x86/kernel/reboot.c
@@ -26,6 +26,8 @@
 # include <asm/iommu.h>
 #endif
 
+#include <acpi/acpi.h>
+
 /*
  * Power off function, if any
  */
@@ -281,6 +283,10 @@ static struct dmi_system_id __initdata r
 
 static int __init reboot_init(void)
 {
+	if (!(acpi_gbl_FADT.boot_flags & ACPI_FADT_8042)) {
+		printk(KERN_INFO "Use ACPI reboot mode\n");
+		reboot_type = BOOT_ACPI;
+	}
 	dmi_check_system(reboot_dmi_table);
 	return 0;
 }
